### PR TITLE
add new module na_cdot_snapshot

### PR DESCRIPTION
**Pull request by marratj**
##### SUMMARY
This PR adds a new module 'na_cdot_snapshot' to create and delete Snapshot Copies on NetApp ONTAP Volumes.

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
na_cdot_snapshot

##### ANSIBLE VERSION
2.4.0

##### ADDITIONAL INFORMATION
This is a simple new module for creating and deleting NetApp ONTAP Snapshot Copies.

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (gouthampacha) [inline lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: This isn't necessary, since it isn't used in this module

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (gouthampacha) [inline lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: Why is this check necessary, since you're querying by self.name and self.volume_name, you'll only get the snapshots?

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (gouthampacha) [inline lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: 215-216 can be changed into `elif self.state == 'present':`

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (gouthampacha) [inline lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: 222, 223 and 224 can be changed into `elif self.state == 'present' and not snapshot_exists:`

**Review by gouthampacha — CHANGES_REQUESTED** [review]
Thanks for your contribution. Some comments noted inline.

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (marratj) [reply lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: True, I will remove it. Was an unnoticed leftover from when I copied the class initially from the na_cdot_volume module.

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (marratj) [reply lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: Right, was also a leftover from the copied module.

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (marratj) [reply lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: You're right, as I'm directly querying the volume, I already know that this matches when I get a result back.

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (marratj) [reply lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: Yes, will change that.

**Comment by lmprice** [comment]
@gouthampacha It looks fine to me. I'll wait for your approval before signing off.

**Review by gouthampacha — APPROVED** [review]
Nice! Thank you for the changes and your contribution!

**Comment by gouthampacha** [comment]
@lmprice: +1 am good with these changes.

**Comment by lmprice** [comment]
And they look fine to me as well. 
shipit

**Comment by gouthampacha** [comment]
shipit

**Comment by jctanner** [comment]
bot_status

File: lib/ansible/modules/storage/netapp/na_cdot_snapshot.py
Comment (gouthampacha) [inline lib/ansible/modules/storage/netapp/na_cdot_snapshot.py]: Guess you'll have to drop this line

**Comment by schmots1** [comment]
close_me
Duplicate of official module

**Pull request closed without merging**
